app/controllers/api/v2/search_references_controller.rb

```diff
@@ -2,15 +2,17 @@ module Api
   module V2
     class SearchReferencesController < ApiController
       def index
-        @search_references = SearchReference.for_letter(letter).by_title.all
+        search_references = letter.present? ? SearchReference.for_letter(letter) : SearchReference
```

Comment:
even only this:
`search_references = SearchReference.for_letter(letter)`
would work,
but if the letter is not present it's better just to refer to `SearchReference` for better performaces.

Reply:
The negative branch of this isn't actually the same kind of thing and the name doesn't make sense.

I'd probably just move this and the `by_title.all` to its own method

```ruby
def search_references
  starting_query = letter.present? ? SearchReference.for_letter(letter) : SearchReference

  starting_query.by_title.all
end
```

Reply:
I also think its valid that if the user does not specify a letter the for_letter method returns just a dataset

Reply:
I added that condition for performance reasons, but I can see that we only save a small fraction of a second, so it's not worthy.
So, let's make the model return the right search_references.

Reply:
which is the default behaviour after all